sdk/maps/maps-search/review/maps-search.api.md

```diff
@@ -0,0 +1,756 @@
+    top?: number;
+}
+
+// @public (undocumented)
+export interface ReverseSearchCrossStreetAddressResult {
+    readonly numResults: number;
+    readonly queryTime: number;
+    readonly results: ReverseSearchCrossStreetAddressResultItem[];
+}
+
+// @public (undocumented)
+export interface ReverseSearchCrossStreetAddressResultItem {
+    readonly address?: Address;
+    readonly position?: LatLon;
+}
+
+// @public
+export type RoadUseType = string;
+
+// @public
+export interface SearchAddressBaseOptions extends SearchBaseOptions {
+    boundingBox?: BoundingBox;
+    isTypeAhead?: boolean;
+    radiusInMeters?: number;
+}
+
+// @public
+export type SearchAddressBatchOptions = OperationOptions & BatchPollerOptions;
+
+// @public
+export interface SearchAddressOptions extends SearchAddressBaseOptions {
+    coordinates?: LatLon;
+    countryCodeFilter?: string[];
+    entityType?: GeographicEntityType;
+}
+
+// @public
+export interface SearchAddressRequest {
+    // (undocumented)
+    options?: SearchAddressOptions;
+    // (undocumented)
+    query: string;
+}
+
+// @public
+export interface SearchAddressResult {
+    readonly fuzzyLevel?: number;
+    readonly geoBias?: LatLon;
+    readonly numResults: number;
+    readonly query?: string;
+    readonly queryTime: number;
+    readonly queryType?: QueryType;
+    readonly results: SearchAddressResultItem[];
+    readonly skip?: number;
+    readonly totalResults?: number;
+}
+
+// @public
+export interface SearchAddressResultItem {
+    readonly address: Address;
+    readonly addressRanges?: AddressRanges;
+    readonly dataSources?: DataSource;
+    readonly detourTime?: number;
+    readonly distanceInMeters?: number;
+    readonly entityType?: GeographicEntityType;
+    readonly entryPoints?: EntryPoint[];
+    readonly id: string;
+    readonly info?: string;
+    readonly matchType?: MatchType;
+    readonly pointOfInterest?: PointOfInterest;
+    readonly position: LatLon;
+    readonly score: number;
+    readonly type: SearchAddressResultType;
+    readonly viewport: BoundingBox;
+}
+
+// @public
+export type SearchAddressResultType = string;
+
+// @public
+export interface SearchAlongRouteOptions extends SearchGeometryBaseOptions, OperationOptions {
+    brandFilter?: string[];
+    electricVehicleConnectorFilter?: ElectricVehicleConnector[];
+}
+
+// @public
+export interface SearchBaseOptions {
+    extendedPostalCodesFor?: SearchIndexes[];
+    language?: string;
+    localizedMapView?: LocalizedMapView;
+    skip?: number;
+    top?: number;
+}
+
+// @public
+export interface SearchExtraFilterOptions {
+    brandFilter?: string[];
+    categoryFilter?: number[];
+    electricVehicleConnectorFilter?: ElectricVehicleConnector[];
+}
+
+// @public (undocumented)
+export type SearchGeometry = GeoJsonPolygon | GeoJsonPolygonCollection | GeoJsonCircleOrPolygonFeatureCollection;
```

Comment:
Can you document the types and resolve all `(undocumented)` that's worth doing?

Reply:
Thanks for pointing this out! I've added the documents which I think is necessary in 979cae726ea82a1e37da239f656293e91e76f414. Most of the rest are GeoJson, and we want the user to refer to the [GeoJson Spec](https://www.rfc-editor.org/rfc/rfc7946) directly.